### PR TITLE
Update comments around deprecated and inclusive naming

### LINKED ISSUE #72: [FEATURE] Add support for inclusive naming
Add support for inclusive naming by deprecating the master nomenclature.

See meta issue https://github.com/opensearch-project/opensearch-clients/issues/16 for details.

### PR DESCRIPTION
**Pull request by harshavamsi**
Signed-off-by: Harsha Vamsi Kalluri <harshavamsi096@gmail.com>

### Description
- Updates `master_timeout` to `cluster_manager_timeout`
- Updates `master` to `cluster_manager`
- Bumps versions to `2.1.0`

### Issues Resolved
Closes #72 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by nhtruong — DISMISSED** [review]
🚀

File: opensearch-api/lib/opensearch/api/actions/snapshot/clone.rb [hunk 1]
```diff
@@ -33,7 +33,8 @@ module Actions
         # @option arguments [String] :repository A repository name
         # @option arguments [String] :snapshot The name of the snapshot to clone from
         # @option arguments [String] :target_snapshot The name of the cloned snapshot to create
-        # @option arguments [Time] :master_timeout Explicit operation timeout for connection to master node
+        # @option arguments [Time] :master_timeout (DEPRECATED: use cluster_manager_timeout instead) Explicit operation timeout for connection to master node
+        # @option arguments [Time] :cluster_manager_timeout Explicit operation timeout for connection to cluster_manager node
         # @option arguments [Hash] :headers Custom HTTP headers
         # @option arguments [Hash] :body The snapshot clone definition (*Required*)
         #
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/actions/snapshot/clone.rb:40]: Don't forget to update the ParamsRegristry for all affected methods.

Comment (harshavamsi) [reply opensearch-api/lib/opensearch/api/actions/snapshot/clone.rb:40]: @nhtruong I've updated them all I believe. Can you take a look?

File: opensearch-api/lib/opensearch/api/actions/cluster/delete_component_template.rb [hunk 2]
```diff
@@ -32,7 +32,8 @@ module Actions
         #
         # @option arguments [String] :name The name of the template
         # @option arguments [Time] :timeout Explicit operation timeout
-        # @option arguments [Time] :master_timeout Specify timeout for connection to master
+        # @option arguments [Time] :master_timeout (DEPRECATED: use cluster_manager_timeout instead) Specify timeout for connection to master
+        # @option arguments [Time] :cluster_manager_timeout Specify timeout for connection to cluster_manager
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/actions/cluster/delete_component_template.rb:36]: Need to update the ParamsRegistry here as well

File: opensearch-api/lib/opensearch/api/actions/cluster/pending_tasks.rb [hunk 3]
```diff
@@ -31,8 +31,9 @@ module Actions
         # Returns a list of any cluster-level changes (e.g. create index, update mapping,
         # allocate or fail shard) which have not yet been executed.
         #
-        # @option arguments [Boolean] :local Return local information, do not retrieve the state from master node (default: false)
-        # @option arguments [Time] :master_timeout Specify timeout for connection to master
+        # @option arguments [Boolean] :local Return local information, do not retrieve the state from cluster_manager node (default: false)
+        # @option arguments [Time] :master_timeout (DEPRECATED: use cluster_manager_timeout instead) Specify timeout for connection to master
+        # @option arguments [Time] :cluster_manager_timeout Specify timeout for connection to cluster_manager
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/actions/cluster/pending_tasks.rb:36]: Missing ParamsRegistry update in this file

File: opensearch-api/lib/opensearch/api/actions/cluster/put_component_template.rb [hunk 4]
```diff
@@ -33,7 +33,8 @@ module Actions
         # @option arguments [String] :name The name of the template
         # @option arguments [Boolean] :create Whether the index template should only be added if new or can also replace an existing one
         # @option arguments [Time] :timeout Explicit operation timeout
-        # @option arguments [Time] :master_timeout Specify timeout for connection to master
+        # @option arguments [Time] :master_timeout (DEPRECATED: use cluster_manager_timeout instead) Specify timeout for connection to master
+        # @option arguments [Time] :cluster_manager_timeout Specify timeout for connection to cluster_manager
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/actions/cluster/put_component_template.rb:37]: Missing ParamsRegistry update in this file

File: opensearch-api/lib/opensearch/api/actions/dangling_indices/import_dangling_index.rb [hunk 5]
```diff
@@ -33,7 +33,8 @@ module Actions
         # @option arguments [String] :index_uuid The UUID of the dangling index
         # @option arguments [Boolean] :accept_data_loss Must be set to true in order to import the dangling index
         # @option arguments [Time] :timeout Explicit operation timeout
-        # @option arguments [Time] :master_timeout Specify timeout for connection to master
+        # @option arguments [Time] :master_timeout (DEPRECATED: use cluster_manager_timeout instead) Specify timeout for connection to master
+        # @option arguments [Time] :cluster_manager_timeout Specify timeout for connection to cluster_manager
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/actions/dangling_indices/import_dangling_index.rb:37]: **Missing ParamsRegistry update in this file**

File: opensearch-api/lib/opensearch/api/actions/indices/delete_alias.rb [hunk 6]
```diff
@@ -33,7 +33,8 @@ module Actions
         # @option arguments [List] :index A comma-separated list of index names (supports wildcards); use `_all` for all indices
         # @option arguments [List] :name A comma-separated list of aliases to delete (supports wildcards); use `_all` to delete all aliases for the specified indices.
         # @option arguments [Time] :timeout Explicit timestamp for the document
-        # @option arguments [Time] :master_timeout Specify timeout for connection to master
+        # @option arguments [Time] :master_timeout (DEPRECATED: use cluster_manager_timeout instead) Specify timeout for connection to master
+        # @option arguments [Time] :cluster_manager_timeout Specify timeout for connection to cluster_manager
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/actions/indices/delete_alias.rb:37]: Missing ParamsRegistry update in this file

**Review by nhtruong — CHANGES_REQUESTED** [review]

File: opensearch-api/lib/opensearch/api/actions/indices/open.rb [hunk 7]
```diff
@@ -32,7 +32,7 @@ module Actions
         #
         # @option arguments [List] :index A comma separated list of indices to open
         # @option arguments [Time] :timeout Explicit operation timeout
-        # @option arguments [Time] :master_timeout Specify timeout for connection to master
+        # @option arguments [Time] :master_timeout (DEPRECATED: use cluster_manager_timeout instead) Specify timeout for connection to master
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/actions/indices/open.rb:35]: Missing arg doc for cluster_manager_timeout

File: opensearch-api/lib/opensearch/api/actions/snapshot/delete.rb [hunk 8]
```diff
@@ -32,7 +32,7 @@ module Actions
         #
         # @option arguments [String] :repository A repository name
         # @option arguments [String] :snapshot A snapshot name
-        # @option arguments [Time] :master_timeout Explicit operation timeout for connection to master node
+        # @option arguments [Time] :master_timeout (DEPRECATED: use cluster_manager_timeout instead) Explicit operation timeout for connection to cluster_manager node
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/actions/snapshot/delete.rb:35]: Missing arg doc for cluster_manager_timeout

File: opensearch-api/lib/opensearch/api/utils.rb [hunk 9]
```diff
@@ -97,13 +97,13 @@ def __pathify(*segments)
...
-      #       { :index =>  { :_index => 'myindexA', :_type => 'mytype', :_id => '1', :data => { :title => 'Test' } } },
-      #       { :update => { :_index => 'myindexB', :_type => 'mytype', :_id => '2', :data => { :doc => { :title => 'Update' } } } }
+      #       { :index =>  { :_index => 'myindexA', :_id => '1', :data => { :title => 'Test' } } },
+      #       { :update => { :_index => 'myindexB', :_id => '2', :data => { :doc => { :title => 'Update' } } } }
       #     ]
       #
-      #     # => {"index":{"_index":"myindexA","_type":"mytype","_id":"1"}}
+      #     # => {"index":{"_index":"myindexA","_id":"1"}}
       #     # => {"title":"Test"}
-      #     # => {"update":{"_index":"myindexB","_type":"mytype","_id":"2"}}
```
Comment (nhtruong) [inline opensearch-api/lib/opensearch/api/utils.rb:106]: Should this change be surfaced in the CHANGELOG.md as well?

**Review by nhtruong — APPROVED** [review]

File: opensearch-api/Rakefile [hunk 10]
```diff
@@ -56,9 +56,9 @@ namespace :test do
       url = ENV['TEST_CLUSTER_URL'] || ENV['TEST_OPENSEARCH_SERVER']
       url = "http://localhost:#{ENV['TEST_CLUSTER_PORT'] || 9200}" unless url
       client = OpenSearch::Client.new :url => url
-      es_version_info = client.info['version']
-      version_number = es_version_info['number']
-      build_hash      = es_version_info['build_hash']
+      os_version_info = client.info['version']
```
Comment (dblock) [inline opensearch-api/Rakefile:59]: We can change this later, but I think this should be `opensearch_` because OS assumes operating system a lot.

**Review by dblock — APPROVED** [review]

**Pull request merged**